source/tdx-guest-hardening.rst

```diff
@@ -1000,4 +1000,325 @@ https://github.com/intel/ccc-linux-guest-hardening/tree/master/bkc/kafl/userspac
+in step 2 needs to be exercised by using either kafl or kfx fuzzers (or both).
+However, if resource or timing is very limited, the fuzzing can be
+primary focused
+only on the 'concern' entries from the step 2 or on any other entries
+that are considered potentially problematic (complex parsing of data, many call
+chains, etc.).
+The typical reported input locations can be divided into two groups:
+driver initialization
+code (init and probe functions) and runtime operation. The first group would be
+the easiest one to reach by a fuzzer since it does not require any
+external stimulus:
+it only requires a creation of a separate fuzzing harness. The second
+one ideally
+requires a functional test suite to be run to exercise the driver
+functionality as a stimulus.
+However, in the absence of such a test suite, a set of simple manual
+tests can be
+created or certain userspace commands/operations performed that trigger
+invocation of the functions reported by smatch in step 2. Setting up
+the driver fuzzing can also be very beneficial even in cases when
+smatch does not report any hits in driver’s init or probe functions,
+because smatch can miss some host input consumption points in some
+cases and fuzzing can help discover such cases.
+
+**Example**. Enabling fuzzing targets like the virtio-vsock driver
+requires some manual work and modifications of the fuzzing setup (as
+opposite to more straightforward examples like virtio-net or
+virtio-console) and below steps explain how to add support for such a
+target. In a nutshell, virtio-vsock sets up a socket on the host/VMM,
+allowing a host process to setup a direct socket connection to the
+guest VM over VirtIO. For fuzzing, this requires some initial setup in
+the host, as well as establishing a connection from the guest.
+
+**Host steps**. First, the VMM host kernel must support VSOCK. The
+corresponding kernel module can be loaded using `modprobe vhost_vsock`.
+If this fails, it might be required to install a
+different kernel which has `CONFIG\_VHOST\_VSOCK` set. When the
+vhost_vsock driver is enabled, a device shall appear at
+`/dev/vhost-vsock`. Its default permissions might be insufficient for
+QEMU to access, but it can be fixed by executing `chmod 0666
+/dev/vhost-vsock`. Now that the vhost-vsock device is available to
+QEMU, the device for the guest VM can be enabled by appending the
+string `-device vhost-vsock-pci,id=vhost-vsock-pci0,guest-cid=3` to
+QEMU options. The guest-cid value is a connection identifier that
+needs to be unique for the system. In other words, when fuzzing with
+multiple workers, each QEMU instance must use a separate guest-cid.
+For kAFL we have added some syntax magic to allow for these
+kinds of situations. In your `kafl_config.yaml` (by default found in
+`$BKC_ROOT/bkc/kafl/kafl_config.yaml`),  the following string can be
+appended to the `qemu_base` entry: `-device vhost-vsock-pci,id=vhost-vsock-pci0,guest-cid={QEMU_ID + 3}`.
```

Comment:
I haven't implemented the {QEMU_ID + 3} thing yet. Will try to push this out this week.